src/Microsoft.ML.Data/DataLoadSave/Text/TextDataSaver.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using System.IO;
+
+namespace Microsoft.ML.Data.IO
+{
+    public static class TextDataSaver
```

Comment:
>TextDataSaver [](start = 24, length = 13)

So, we have this class, that calls `TextSaver`, we have the other class that calls `BinarySaver`... what really would have been lost if we'd just made these static methods on the classes themselves, and called it a day?

Reply:
Just to be clear, we already have I know a fair number of convenience methods on the binary guys for just this sort of thing.

---
In reply to: [214523430](https://github.com/dotnet/machinelearning/pull/798#discussion_r214523430) [](ancestors = 214523430)

Reply:
I do not mind doing that. However, `BinarySaver/TextSaver` are currently in the `Runtime` namespace, and these guys are in the 'blessed official' namespace.

Also, I would prefer not to expose `XXXSaver.Arguments` as a constructor parameter. 
So, either we should really rework `BinarySaver/TextSaver` to become proper API members or hide them completely and expose only a pair of convenience saving methods (the route I've taken in this PR)

Reply:
Actually, I like the idea that we do NOT have arguments to savers. Let us save in the format we ourselves can read back with full fidelity, and not be in the business of customizing save output.

Reply:
> Actually, I like the idea that we do NOT have arguments to savers. Let us save in the format we ourselves can read back with full fidelity,

Wait, what does the second statement have to do with the first? Assuredly configuration is useful, and has nothing to do with fidelity. Indeed I have trouble thinking of a situation where I've ever configured a saver, and the data has not been full fidelity (assuming, of course, the saver actually has a corresponding reader). Forcing the text saver to be dense is pretty useful. Forcing a binary saver to not use a compression algorithm can be useful. Forcing a saver to save hidden columns is useful. These have nothing to do with fidelity of the data that is written -- there are simply multiple ways of writing the data, which are each situationally useful.

Configuring save output is really, *really* useful.